examples/serial/Makefile

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (C) 2009-2012 Chris McClelland
+# Copyright (C) 2015 Joel Stanley <joel@jms.id.au>
+# Copyright (C) 2017 Kyle Robbertze <krobbertze@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# To build the firmware you will need:
+#   SDCC from http://sdcc.sourceforge.net
+#   fx2lib from http://fx2lib.wiki.sourceforge.net
+#
+TARGET := serial
+FX2LIBDIR := ../../third_party/fx2lib
+LIBS := $(FX2LIBDIR)/lib/fx2.lib
+INCS := -I sdcc -I$(FX2LIBDIR)/include -I.
+
+USE_16K := 1
+
+CC_SRCS := serial.c
+CC_OBJS := $(CC_SRCS:%.c=%.rel)
+
+CC := sdcc
+
+CFLAGS += -mmcs51 --xram-size 0x0200 $(FLAGS)
+
+ifeq ($(USE_16K),1)
+	CFLAGS += --code-size 0x3e00
+else
+	CFLAGS += --code-size 0x1e00
+endif
+
+# Use make V=1 for a verbose build.
+ifndef V
+	Q_CC=@echo      '      CC ' $@;
+	Q_AS=@echo      '      AS ' $@;
+	Q_LINK=@echo    '    LINK ' $@;
+	Q_RM=@echo      '   CLEAN ';
+	Q_OBJCOPY=@echo ' OBJCOPY ' $@;
+	Q_GEN=@echo     '     GEN ' $@;
+endif
+
+.PHONY: all clean
+
+all: $(TARGET).hex
+
+$(CC_SRCS): $(FX2LIBDIR)/lib/fx2.lib
+
+$(FX2LIBDIR)/lib/fx2.lib: $(FX2LIBDIR)/.git
+	cd $(dir $@) && make -j1
+
+# We depend on a file inside the directory as git creates an
```

Comment:
probably be a good idea. Would be nice to just `include` something instead of copying it every time